addons/mail/models/mail_channel.py

```diff
@@ -115,6 +116,14 @@ def _compute_avatar_128(self):
         for record in self:
             record.avatar_128 = record.image_128 or record._generate_avatar()
 
+    @api.depends('channel_type', 'is_member')
+    def _compute_is_avatar_editable(self):
+        for channel in self:
+            if channel.channel_type == 'channel':
+                channel.is_avatar_editable = self.sudo(False).env.is_admin() or channel.create_uid.id == self.env.user.id
```

Comment:
I don't think this is correct. We use a channel together and I can change the avatar from settings page without being the creator of the channel (nor the admin of DB).

The same rules for editing avatar from settings page should apply here.

Reply:
I think this is what I did in this case, otherwise, it is not needed to use a field in python to check the rights for editing. If you log in as Demo and you cannot change the avatar through the settings page.

Reply:
Ok, didn't saw the changes in channel settings, so both matches same condition (only allow admin to change avatar)